### PR TITLE
update nx version to 12.4.0

### PR DESCRIPTION
**Pull request by xiongemi**
- update nx version of the workspace to 12.4.0

### CODE REVIEW

**Comment by xiongemi** [comment]
failed due to react native and jest 27 issue: https://github.com/callstack/react-native-testing-library/issues/743.
@nrwl/jest will migrate jest to 27.0.3. however, it has issues with react native testing.

**Comment by xiongemi** [comment]
upgrade react naive version to 0.65. however, the unit test does not run because @testing-library/react-native does not support react native version 0.65. got a timeout error when i try to run unit tests: `thrown: "Exceeded timeout of 5000 ms for a hook.` on line `@testing-library/react-native`.
convert this to a draft or.
TODO:
- add migration script to add `testEnvironment: 'node'` to all jest.config for react native app and lib
-  add migration script to add react-native-codegen to dev dependency

File: jest.config.js [hunk 1]
```diff
@@ -1,9 +1,6 @@
 module.exports = {
-  testMatch: ['**/+(*.)+(spec|test).+(ts|js)?(x)'],
-  transform: {
-    '^.+\\.(ts|js|html)$': 'ts-jest',
-  },
-  resolver: '@nrwl/jest/plugins/resolver',
-  moduleFileExtensions: ['ts', 'js', 'html'],
-  coverageReporters: ['html'],
+  projects: [
```
Comment (jaysoo) [inline jest.config.js:2]: The new `@nrwl/jest` package exports a `getJestProjects() function that can be used here.

```
projects: getJestProjects()
```

**Review by jaysoo — CHANGES_REQUESTED** [review]

File: jest.config.js (hunk 1 above)
Comment (xiongemi) [reply jest.config.js]: done

**Pull request merged**